docs/earthly-command/earthly-command.md

```diff
@@ -646,6 +646,130 @@ List secrets the current account has access to.
+
+* ```
+  earthly registry setup --username <username> --password <password> [<host>]
+  earthly registry setup --org <org> --project <project> --username <username> --password <password> [<host>]
+  ```
+
+##### AWS elastic container registry (`--cred-helper=ecr-login`)
+
+* ```
+  earthly registry setup --cred-helper ecr-login --aws-access-key-id <key> --aws-secret-access-key <secret> <host>
+  earthly registry setup --org <org> --project <project> --cred-helper ecr-login --aws-access-key-id <key> --aws-secret-access-key <secret> <host>
+  ```
+
+##### GCP artifact or container registry (`--cred-helper=gcloud`)
+
+* ```
+  earthly registry setup --cred-helper gcloud --gcp-key <key> <host>
+  earthly registry setup --org <org> --project <project> --cred-helper gcloud --gcp-key <key> <host>
+  ```
+
+#### Description
+
+Store registry credentials in the earthly-cloud secrets store. These credentials are used to authenticate with the registry.
+When they are associated with a project, by specifying `--org`, and `--project` flags, they will be associated with the project (as referenced by the
+`PROJECT` Earthfile command), which is used when running in CI.
+
+{% hint style='info' %}
+##### Note
+Registry credentials are stored under `std/registry/<host>/...` of either the user, or project based secrets.
+
+The `earthly registry ...` commands exist for convience; however, it is possible to set (or delete) these values using the `earthly secrets ...` commands.
+{% endhint %}
+
+
+#### Options
+
+##### `--cred-helper`
+
+When specified, use a credential helper for authenticating with the registry. Values can be `ecr-login`, `gcloud`, or `none`.
+
+Also available as an env var setting: `EARTHLY_REGISTRY_CRED_HELPER=<value>`.
+
+##### `--username <username>`
+
+The username to use; only applicable when `--cred-helper` is omitted (or `none`).
+
+Also available as an env var setting: `EARTHLY_REGISTRY_USERNAME=<value>`.
+
+##### `--password <password>`
+
+The password to use; only applicable when `--cred-helper` is omitted (or `none`).
+
+Also available as an env var setting: `EARTHLY_REGISTRY_PASSWORD=<value>`.
+
+##### `--password-stdin`
+
+When set, read the password from stdin; only applicable when `--cred-helper` is omitted (or `none`).
+
+Also available as an env var setting: EARTHLY_REGISTRY_PASSWORD_STDIN=true.
+
+##### `--aws-access-key-id <identifier>`
+
+The AWS access key ID to use when requesting a registry token, only applicable when `--cred-helper=ecr-login`.
+
+Also available as an env var setting: `AWS_ACCESS_KEY_ID=<identifier>`.
+
+##### `--aws-secret-access-key <secret>`
+
+The AWS secret access key to use when requesting a registry token, only applicable when `--cred-helper=ecr-login`.
+
+Also available as an env var setting: `AWS_SECRET_ACCESS_KEY=<secret>`.
+
+##### `--gcp-key <key>`
+
+The GCP key to use when requesting a registry token, only applicable when `--cred-helper=gcloud`.
```

Comment:
similarly to my comment in the implementation PR, perhaps it would be good to expand on what this key is, I could be wrong but I'm concerned it might not be as clear as the AWS access key and secret.